src/models/blockchain/transaction/transaction_kernel.rs

```diff
@@ -98,6 +101,25 @@ impl From<PrimitiveWitness> for TransactionKernel {
     }
 }
 
+impl TransactionKernel {
+    /// Determine if the transaction double spends in input, either by spending an input that was
+    /// already spent, or by attempting to spend the same input multiple times. Relative to a given
+    /// mutator set. Will also return true if any removal records have invalid membership proofs. So
+    /// the caller should check for that before calling this method.
```

Comment:
why should caller check it?   that sounds like duplicate work.   I believe either caller should check it, or this method, not both.